### DISCUSSION TITLE
请教一下paopaoDNS和paopaogateway的协同问题

### DISCUSSION THREAD

**Discussion by BiGnotes** [kkkgo/PaoPaoGateWay#133] [问和答]
#在PVE上用ct+docker的方式搭建了DNS，IP192.168.1.4
用虚拟机搭建GATEWAY,IP192.168.1.3
用nslookup访问国内域名没问题，但是国外域名一直没有结果，还请大佬帮忙看一下。

![QQ截图20241007231117](https://github.com/user-attachments/assets/852711a8-0793-4b57-bf11-f0e81a26666d)
![QQ截图20241007231500](https://github.com/user-attachments/assets/59ffabce-0ccd-4c36-9294-6797448d1bab)

dns的配置如下
docker run -d --name paopaodns -v ~/paopaodns/data:/data -e CNAUTO=yes -e DNS_SERVERNAME=Main.DNS -e SERVER_IP=192.168.1.4 -e IPV6=yes_only6 -e CUSTOM_FORWARD=192.168.1.3:53 -e AUTO_FORWARD=yes -e AUTO_FORWARD_CHECK=yes -e HTTP_FILE=yes -e USE_MARK_DATA=yes --restart always -p 53:53/tcp -p 53:53/udp -p 7889:7889/tcp -p 5304:5304/tcp -p 5304:5304/udp sliamb/paopaodns

gateway的ppgw.ini如下

`#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: free
mode=suburl

# Set fakeip's CIDR here
# default: fake_cidr=7.0.0.0/8
fake_cidr=7.0.0.0/8

# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.1.4
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304

# Clash's web dashboard
clash_web_port="80"
clash_web_password="clashpass"

# default：openport=no
# socks+http mixed 1080
openport=yes

# default: udp_enable=no
udp_enable=no

# default:30
sleeptime=30

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="10.10.10.5"
# default: socks5_port="7890"
socks5_port="7890"

# ovpn mode settting
# The ovpn file in the same directory as the ppgw.ini.
# default: ovpnfile=custom.ovpn
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

# yaml mode settting
# The yaml file in the same directory as the ppgw.ini.
# default: yamlfile=custom.yaml
yamlfile="custom.yaml"

# suburl mode settting
suburl="XXXXXXXXXXXXXXXXXXXXXXXXXX"
subtime=1d

# fast_node=check/yes/no
fast_node=yes
test_node_url="https://www.youtube.com/generate_204"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"
fall_direct="no"
# dns burn setting
# depend on fast_node=yes & mode=suburl/yaml
dns_burn=no
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="192.168.1.4"

# Network traffic records
net_rec=no
max_rec=5000
`

**Comment by kkkgo** [comment]
![image](https://github.com/user-attachments/assets/649f149c-4de5-4935-a1cb-12ad08892018)
你的CPU或者pve配置不支持v3架构但你定制的clash二进制是v3架构。

**Reply by BiGnotes** [reply]
谢谢大佬，CPU是 Celeron(R) J4125，那就是用错了clash的版本，那我再试一下，这条消息没标红，以为就是简单的提示了。。

**Reply by kkkgo** [reply]
<img width="525" alt="image" src="https://github.com/user-attachments/assets/c74b7922-c015-4c2f-9426-68e7207d9b90">

Usually you should select for your VM a processor type which closely matches the CPU of the host system, as it means that the host CPU features (also called CPU flags ) will be available in your VMs. If you want an exact match, you can set the CPU type to host in which case the VM will have exactly the same CPU flags as your host system.
通常，您应该为 VM 选择与主机系统的 CPU 紧密匹配的处理器类型，因为这意味着主机 CPU 功能（也称为 CPU 标志）将在您的 VM 中可用。如果需要完全匹配，可以将 CPU 类型设置为 host，在这种情况下，VM 将具有与主机系统完全相同的 CPU 标志。

**Reply by BiGnotes** [reply]
我换成了mihomo-linux-amd64-compatible 版本就可以了，现在nslookup已经可以获取IP了

还有个问题想一并请教一下
主路由是openwrt，请问静态路由应该怎么设置，下拉有好多种类型，对于小白有点懵。。。

![QQ截图20241007234124](https://github.com/user-attachments/assets/8a6a4b4e-201d-40f4-848d-9f7768ff36cb)
![QQ截图20241007234130](https://github.com/user-attachments/assets/20e175ae-a4c4-43b4-bc2a-966ed7eae421)

**Reply by BiGnotes** [reply]
> <img alt="image" width="525" src="https://private-user-images.githubusercontent.com/4530897/374233548-c74b7922-c015-4c2f-9426-68e7207d9b90.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzMTYyMjksIm5iZiI6MTcyODMxNTkyOSwicGF0aCI6Ii80NTMwODk3LzM3NDIzMzU0OC1jNzRiNzkyMi1jMDE1LTRjMmYtOTQyNi02OGU3MjA3ZDliOTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDdUMTU0NTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDA3YTMxMjU2Njk2NWQzYTE3OWE0ZjEyOTVjNTk5ZWUzNDYwNjE3MjljZGMzMzEzNmIzNWQ5NThhMThjOTEwMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.nLF6rI6OmnVNKE9xAu4TBVmAkILsy2q8w1W-9rypV-A">
> Usually you should select for your VM a processor type which closely matches the CPU of the host system, as it means that the host CPU features (also called CPU flags ) will be available in your VMs. If you want an exact match, you can set the CPU type to host in which case the VM will have exactly the same CPU flags as your host system. 通常，您应该为 VM 选择与主机系统的 CPU 紧密匹配的处理器类型，因为这意味着主机 CPU 功能（也称为 CPU 标志）将在您的 VM 中可用。如果需要完全匹配，可以将 CPU 类型设置为 host，在这种情况下，VM 将具有与主机系统完全相同的 CPU 标志。

这个我也查了一下，kvm64只有基础的指令集，所以应该用compatible版本，如果改成host的话，应该是可以用完整的版本的clash.meta，不过好像也没啥太多必要。

> pve默认使用一种叫KVM64类似奔腾4的CPU模型。这个模型只有基础功能。

> 因此kvm64实际上是一个缺少指令集，如果需要特定的指令集，需要在虚拟配置文件中，加上指令集，例如aes。

**Reply by kkkgo** [reply]
cpu type选host应该能支持v3。
静态路由类型默认就行，接口选lan。
